resip/recon/ConversationManager.hxx

```diff
@@ -483,6 +486,18 @@ public:
    virtual void onDtmfEvent(ParticipantHandle partHandle, int dtmf, int duration, bool up) = 0;
 
    ///////////////////////////////////////////////////////////////////////
+   // Pager Message Handlers /////////////////////////////////////////////
+   ///////////////////////////////////////////////////////////////////////
+
+   virtual void onMessageArrived(resip::ServerPagerMessageHandle handle, const resip::SipMessage& message) = 0;
+   // Called when a MESSAGE has been successfully sent
+   virtual void onSuccess(resip::ClientPagerMessageHandle handle, const resip::SipMessage& status) = 0;
+   // Application could re-page the failed contents or just ingore it.
+   virtual void onFailure(resip::ClientPagerMessageHandle handle, const resip::SipMessage& status, std::auto_ptr<resip::Contents> contents) = 0;
+   
+   const char* sendMessage(const resip::NameAddr& destination, const resip::Data& msg, const resip::Mime& mimeType);
```

Comment:
@sgodin do you mean the existing ```Conversation``` object though, or should some other ```Conversation``` object be used for chat?

E.g. we could have an abstract ```Conversation``` and then subclasses ```ChatConversation``` and ```MediaConversation```

The distinction between the two may be the existence of a SIP dialog.

Given that conversations can change (e.g. they start on chat, then do some webcam call, then fall back to chat for a while), maybe it is better not to have such a class hierarchy though.

When recon is used on the desktop (which is the aim here), the user may have a chat conversation with one peer and a combined chat/webcam session with another peer concurrently, but those would be 2 or 3 different Conversation instances.

If you are not happy for this to go into the API, what would you feel is a reasonable bare-minimum implementation of this that we could accept in recon for now?  I can also spend some time helping on the API side over the next few days while Mateus finishes the telepathy component.